src/main/java/com/conveyal/datatools/manager/jobs/MergeLineContext.java

```diff
@@ -0,0 +1,899 @@
+        skipFile = shouldSkipFile();
+        if (csvReader == null) {
+            skipFile = true;
+            LOG.warn("Table {} not found in the zip file for {}{}", table.name, feedSource.name, version.version);
+            return;
+        }
+        fieldsFoundInZip = table.getFieldsFromFieldHeaders(csvReader.getHeaders(), null);
+        fieldsFoundList = Arrays.asList(fieldsFoundInZip);
+        // Determine the index of the key field for this version's table.
+        keyFieldIndex = getFieldIndex(fieldsFoundInZip, keyField);
+        if (keyFieldIndex == -1) {
+            LOG.error("No {} field exists for {} table (feed={})", keyField, table.name, version.id);
+            keyFieldMissing = true;
+            // If there is no agency_id for agency table, create one and ensure that
+            // route#agency_id gets set.
+        }
+    }
+
+    public boolean shouldSkipFile() {
+        if (handlingActiveFeed && job.mergeType.equals(SERVICE_PERIOD)) {
+            // Always prefer the "future" file for the feed_info table, which means
+            // we can skip any iterations following the first one.
+            return EXTEND_FUTURE.equals(mergeFeedsResult.mergeStrategy) || table.name.equals("feed_info");
+        }
+        return false;
+    }
+
+    /**
+     * Iterate over all rows in table and write them to the output zip.
+     * @return false, if a failing condition was encountered. true, if everything was ok.
+     */
+    public boolean iterateOverRows() throws IOException {
+        // Iterate over rows in table, writing them to the out file.
+        while (csvReader.readRecord()) {
+            startNewRow();
+            if (checkMismatchedAgency()) {
+                // If there is a mismatched agency, return immediately.
+                return false;
+            }
+            updateFutureFeedFirstDate();
+            // If checkMismatchedAgency flagged skipFile, loop back to the while loop. (Note: this is
+            // intentional because we want to check all of the agency ids in the file).
+            if (skipFile || lineIsBlank()) continue;
+            // Check certain initial conditions on the first line of the file.
+            checkFirstLineConditions();
+            initializeRowValues();
+            // Construct row values. If a failure condition was encountered, return.
+            if (!constructRowValues()) {
+                return false;
+            }
+            finishRowAndWriteToZip();
+        }
+        return true;
+    }
+
+    public void startNewRow() throws IOException {
+        keyValue = csvReader.get(keyFieldIndex);
+        // Get the spec fields to export
+        List<Field> specFields = table.specFields();
+        // Filter the spec fields on the set of fields found in all feeds to be merged.
+        sharedSpecFields = specFields.stream()
+            .filter(f -> containsField(allFields, f.name))
+            .collect(Collectors.toList());
+    }
```

Comment:
Perhaps this is also a candidate for a [Extract Subclass](https://refactoring.guru/extract-subclass) refactor where a RecordContext class could be created.